### PR TITLE
chore: Clarify note about code building on `sam local start-api`

### PR DESCRIPTION
**Pull request by yuyokk**
Hi team,

Running code locally with `sam local start-api` and see this message 

> You can now browse to the above endpoints to invoke your functions. You do not need to restart/reload SAM CLI while working on your functions, changes will be reflected instantly/automatically. You only need to restart SAM CLI if you update your AWS SAM template

I assumed (similar to what was described here https://github.com/aws/aws-sam-cli/issues/901 and https://github.com/aws/aws-sam-cli/issues/1921) that my typescript code will be built as well but it is not the case.

Thought of clarifying this in the text of that note.

Please let me know if this makes sense.

Thank you

> By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/local/lib/local_api_service.py [hunk 1]
```diff
@@ -72,7 +72,8 @@ def start(self):
         LOG.info(
             "You can now browse to the above endpoints to invoke your functions. "
             "You do not need to restart/reload SAM CLI while working on your functions, "
-            "changes will be reflected instantly/automatically. You only need to restart "
+            "changes will be reflected instantly/automatically (note: you will still need to "
```
Comment (mildaniel) [inline samcli/commands/local/lib/local_api_service.py:75]: I think there's actually a little more to it than that. This message was correct for certain runtimes before `sam build`. For example, if you're using Node.js or Python without running `sam build` then this message is true. As we evolved it seems like some of our messaging hasn't.

Maybe something along the lines of?
```suggestion
            changes will be reflected instantly/automatically. If you used sam build before running local commands, you will need to re-run sam build for the changes to be picked up. You only need to restart
```

**Review by mildaniel — COMMENTED** [review]
Thanks for bringing this up! It's a great user experience improvement to have more accurate messaging! Left a small comment to discuss some of the wording.

File: samcli/commands/local/lib/local_api_service.py (hunk 1 above)
Comment (yuyokk) [reply samcli/commands/local/lib/local_api_service.py]: @mildaniel this makes sense!

**Review by mildaniel — APPROVED** [review]

**Review by mndeveci — APPROVED** [review]

**Comment by mildaniel** [comment]
Thank you @yuyokk for your contribution!

**Pull request merged**